src/main/java/eu/europeana/metis/sandbox/service/workflow/EnrichmentServiceImpl.java

```diff
@@ -1,13 +1,35 @@
 package eu.europeana.metis.sandbox.service.workflow;
 
+import static java.util.Objects.requireNonNull;
+
+import eu.europeana.enrichment.rest.client.DereferenceOrEnrichException;
+import eu.europeana.enrichment.rest.client.EnrichmentWorker;
+import eu.europeana.metis.sandbox.common.exception.RecordProcessingException;
 import eu.europeana.metis.sandbox.domain.Record;
+import java.io.UnsupportedEncodingException;
+import org.jibx.runtime.JiBXException;
 import org.springframework.stereotype.Service;
 
 @Service
 class EnrichmentServiceImpl implements EnrichmentService {
 
+  private final EnrichmentWorker enrichmentWorker;
+
+  public EnrichmentServiceImpl(EnrichmentWorker enrichmentWorker) {
+    this.enrichmentWorker = enrichmentWorker;
+  }
+
   @Override
   public Record enrich(Record record) {
-    return record;
+    requireNonNull(record, "Record must not be null");
+
+    String result;
+    try {
+      result = enrichmentWorker.process(record.getContent());
+    } catch (DereferenceOrEnrichException | JiBXException | UnsupportedEncodingException e) {
```

Comment:
I am catching all of these exceptions, but under the assupmtion that none of those will be a stopping point for the record, is that correct ?